src/handlers/proposalsHandlers.ts

```diff
@@ -53,26 +55,31 @@ const getProposals = (
 };
 
 const getProposal = (
-	req: Request<{ id: string }>,
+	req: AuthenticatedRequest,
 	res: Response,
 	next: NextFunction,
 ): void => {
-	const id = Number.parseInt(req.params.id, 10);
-	if (Number.isNaN(id)) {
-		next(new InputValidationError('The entity id must be a number.', []));
+	const { user } = req;
+	const { proposalId } = req.params;
+	if (user === undefined) {
```

Comment:
This is a little frustrating. It's already an `AuthenticatedRequest` so why is TypeScript so daft that it needs an additional check? Ah, because we defined `AuthenticatedRequest` loosely, it may or may not have `user` and may or may not have `role`. Would it make sense to tighten that up a bit such that `AuthenticatedRequest` gives a stronger guarantee?

Reply:
Unfortunately that isn't allowed, but for good reason: the problem is that we can't know for sure from the handlers that the middleware actually ran -- so while we could tighten the type, it creates a situation where we lose a little bit of type safety, since there's no contractual guarantee that the type is accurate.

If we try to make it tighter, then TypeScript will (correctly) error when we try to pass the handler to the express route, saying that the handler expects something with a defined user but express is not promising that user will exist.

(you can see this if you remove the `?` in the type and try to run)

Reply:
I see. Yes, I would only want to tighten it up if it is possible to tighten it up. I guess this is the first downside of the fine-grained middleware approach I've seen. And it's not that bad, we just check to see if the expected data are present.